### PR TITLE
Improve mold linker configurations for some corner cases

### PR DESCRIPTION
**Pull request by daschuer**
This checks now if gcc is patched for mold like in Ubuntu Jammy and finds the mold symlink in various places for the -B option as fall back.

### CODE REVIEW

**Comment by ronso0** [comment]
@daschuer I'm still having trouble with the cmake logic, I'm not the right one to review this atm.
@JoergAtGithub or @Holzhaus maybe?

File: CMakeLists.txt [hunk 1]
```diff
@@ -542,58 +542,59 @@ else()
 endif()
 
 if(NOT MSVC)
-  find_program(MOLD_EXECUTABLE mold)
-  default_option(MOLD_SUPPORT "Use 'mold' for linking" "MOLD_EXECUTABLE")
+  # Try to configure mold as linker via -fuse-ld=mold, support from gcc 12.1, gcc 11.2.0-16 or clang
+  execute_process(
```
Comment (Holzhaus) [inline CMakeLists.txt:546]: Maybe use https://cmake.org/cmake/help/latest/module/CheckCompilerFlag.html?

File: CMakeLists.txt [hunk 2]
```diff
@@ -542,58 +542,59 @@ else()
...
+  if(NOT MOLD_FUSE_VERSION_STRING)
+    # check if the symlink ld is in the mold folder for older compiler
+    find_program(MOLD_SYMLINK "${CMAKE_INSTALL_LIBEXECDIR}/mold/ld")
+    get_filename_component(MOLD_SYMLINK_DIRECTORY ${MOLD_SYMLINK} DIRECTORY)
+    set(CMAKE_FIND_DEBUG_MODE FALSE)
+  endif()
+  default_option(MOLD_SUPPORT "Use 'mold' for linking" "MOLD_FUSE_VERSION_STRING OR MOLD_SYMLINK")
   if(MOLD_SUPPORT)
-    if(MOLD_EXECUTABLE)
+    if(MOLD_FUSE_VERSION_STRING)
```
Comment (Holzhaus) [inline CMakeLists.txt:560]: I recommend to use a name that communicates what you are interested in. Something like `MOLD_FUSE_FOUND` maybe?

**Review by Holzhaus — COMMENTED** [review]
I can check the cmake code, but I have no idea about the different variants of selection the proper linker. I just have `mold` executable in `/usr/bin` and that's it.

File: CMakeLists.txt (hunk 1 above)
Comment (daschuer) [reply CMakeLists.txt:546]: The advantage of querying the version via the compiler is that the whole chain is checked and not only the compiler flag. 
check_compiler_flag() does only check the flag.

**Comment by daschuer** [comment]
Done

**Comment by daschuer** [comment]
@Holzhaus Can this be merged now?

**Comment by daschuer** [comment]
ping @Holzhaus

**Comment by daschuer** [comment]
This is ready for merge IMHO

**Comment by Holzhaus** [comment]
Sorry for the long delay, but I intentionally used the comment feature instead of "Request Changes" to indicate that my comments are not a blocker and someone else may merge without a second review loop by me. 

I currently cannot build Mixxx due to #12505. The issue also occurs with this branch. Cmake configuration still works so it's likely okay, but someone else who is able to actually build Mixxx should confirm and merge.

**Comment by ronso0** [comment]
Alright, thanks.
I'll try to find time to remove my mold symlinks in order to test this.

**Comment by ronso0** [comment]
I can built and verify mold was used with
/usr/local/bin/ld.mold symlink mold
`MOLD_SUPPORT` is set TRUE
I renamed the mold binary and updated the symlink, so I assume cmake is using ld.mold

Anything else I should test?

**Comment by daschuer** [comment]
No, thank you for testing. I think we can just merge it and will notice quickly if something falls apart.

**Pull request merged**